tests/unit/plugins/modules/test_flrtvc.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2020- IBM, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import unittest
+from unittest import mock
+import copy
+
+from ansible_collections.ibm.power_aix.plugins.modules import flrtvc
+
+from .common.utils import (
+    rootdir, mock_ftp_report, mock_http_report, mock_https_report
+)
+
+params = {
+        'apar': None,
+        'filesets': None,
+        'csv': None,
+        'path': '/var/adm/ansible',
+        'save_report': False,
+        'verbose': False,
+        'force': False,
+        'clean': False,
+        'check_only': False,
+        'download_only':False,
+        'extend_fs':True,
+        'protocol': None,
+}
+
+init_results = {
+    'changed': False,
+    'msg': '',
+    'meta':{'messages': []}
+    # meta structure will be updated as follow:
+    # meta={'messages': [],     detail execution messages
+    #       '0.report': [],     run_flrtvc reports the vulnerabilities
+    #       '1.parse': [],      run_parser builds the list of URLs
+    #       '2.discover': [],   run_downloader builds the list of epkgs found in URLs
+    #       '3.download': [],   run_downloader builds the list of downloaded epkgs
+    #       '4.1.reject': [],   check_epkgs builds the list of rejected epkgs
+    #       '4.2.check': [],    check_epkgs builds the list of epkgs checking prerequisites
+    #       '5.install': []}    run_installer builds the list of installed epkgs
+}
+
+module = None
+results = None
+mock_https = []
+mock_http = []
+mock_ftp = []
+
+#this function searches each item in a dictionary for a substring.
+#it is used to check the "1.parse" part of "meta" in the report for a protocol.
+#this will determine whether the changes to the code worked as intended.
+def _search_dict(subs, s_dict):
+    for x in range(len(s_dict)):
+        if subs in s_dict[x]:
+            return True
+    return False
+
+#this function is for the case of searching for "http" in a report that has been
+#changed to be "https" because the new protocol string contains the old protocol's string.
+def _instance_count(subs, s_dict):
```

Comment:
since _search_dict and _instance_count effectively serve the same purpose. i think you can use the `re` module to search for a regular expression pattern

if you wanted to go through each item in the 1.parse list and check for the "http", "https", "ftp"
NOTE: i have it s_lst here because s_dict if really a list the way you were using it.
```
def _search_dict(sub, s_lst):
    for url in s_lst:
        # the colon is important as to not match http in https
        pattern = r"^%s:" % subs
        found = re.search(pattern, url)
        if not found:
            # return false if even one of the url does not have the correct protocol
            return False
    # return true if all url in the list uses the specified protocol
    return True
```

Reply:
I definitely agree with you that the re module is better here, thank you, and you're right it was a list ,not a dictionary.